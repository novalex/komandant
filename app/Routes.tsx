import React from 'react';
import { Switch, Route } from 'react-router-dom';

import App from './containers/App';
import HomePage from './containers/HomePage';
import { WrappedLoader } from './components/Loader';

import routes from './constants/routes.json';

// Lazily load routes and code split with webpack
const LazyNewCommandPage = React.lazy(() =>
	import(
		/* webpackChunkName: "NewCommandPage" */
		'./containers/NewCommandPage'
	)
);

const NewCommandPage = (props: Record<string, unknown>) => (
	<React.Suspense fallback={<WrappedLoader />}>
		<LazyNewCommandPage {...props} />
	</React.Suspense>
);

const NoMatch = () => <>Page does not exist</>;

export default function Routes() {
	return (
		<App>
			<Switch>
				<Route exact path={routes.HOME} component={HomePage} />
				<Route path={routes.NEW_COMMAND} component={NewCommandPage} />
				<Route component={NoMatch} />
			</Switch>
		</App>
	);
}
