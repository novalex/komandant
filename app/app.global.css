:root {
	--spacing: 16px;
}

body {
	position: relative;
	color: #eee;
	height: 100vh;
	margin: 0;
	background-color: #222;
	font-family: Arial, Helvetica, Helvetica Neue, serif;
	overflow-y: hidden;
}

h2 {
	margin: 0;
	font-size: 1.75rem;
	letter-spacing: -0.025em;
	color: #fff;
}

p {
	font-size: 24px;
}

li {
	list-style: none;
}

a {
	color: rgb(255, 161, 107);
	text-decoration: none;
	transition: color 0.3s;
}

a:hover {
	color: #fff;
	cursor: pointer;
}

.backButton {
	float: left;
	margin-right: var(--spacing);
}

.backButton a {
	--size: 32px;

	width: var(--size);
	height: var(--size);
	line-height: var(--size);
	text-align: center;
	color: #ddd;
	display: block;
}

.backButton a:hover {
	color: #fff;
}

/*
 * @NOTE: Prepend a `~` to css file paths that are in your node_modules
 *        See https://github.com/webpack-contrib/sass-loader#imports
 */
@import '~@fortawesome/fontawesome-free/css/all.css';
