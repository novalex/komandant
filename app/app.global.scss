:root {
	--spacing: 16px;
	--color: #eee;
	--background: #1a1a1a;
	--border-color: #000;
	--header-border: rgba(0, 0, 0, 0.15);
	--header-background: #111;
	--header-height: 42px;
}

* {
	box-sizing: border-box;
}

body {
	position: relative;
	color: var(--color);
	height: 100vh;
	margin: 0;
	background-color: var(--background);
	font-family: Arial, Helvetica, Helvetica Neue, serif;

	/* overflow-y: hidden; */
}

#root {
	height: 100vh;
}

h2 {
	margin: 0;
	font-size: 1.75rem;
	letter-spacing: -0.025em;
	color: #fff;
}

li {
	list-style: none;
}

a {
	color: rgb(255, 161, 107);
	text-decoration: none;
	transition: color 0.3s;

	&:hover {
		color: #fff;
		cursor: pointer;
	}
}

/*
 * @NOTE: Prepend a `~` to css file paths that are in your node_modules
 *        See https://github.com/webpack-contrib/sass-loader#imports
 */
@import '~@fortawesome/fontawesome-free/css/all.css';
