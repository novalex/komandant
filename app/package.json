{
  "name": "komandant-app",
  "productName": "Komandant",
  "version": "1.0.0",
  "description": "App for saving and executing CLI commands via friendly UI.",
  "main": "./main.prod.js",
  "author": {
    "name": "Alex I. Nitu",
    "email": "alex@novalx.com",
    "url": "https://github.com/novalex"
  },
  "scripts": {
    "electron-rebuild": "node -r ../internals/scripts/BabelRegister.js ../internals/scripts/ElectronRebuild.js",
    "postinstall": "yarn electron-rebuild"
  },
  "license": "MIT",
  "dependencies": {}
}
