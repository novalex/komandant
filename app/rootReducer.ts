import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';
import { History } from 'history';
// eslint-disable-next-line import/no-cycle
import newCommandReducer from './features/new-command/newCommandSlice';

export default function createRootReducer(history: History) {
	return combineReducers({
		router: connectRouter(history),
		newCommand: newCommandReducer,
	});
}
