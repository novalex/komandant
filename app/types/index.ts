export type CommandArgument = {
	find: string;
	replace: string;
};

export type CommandItem = {
	id: string;
	name: string;
	content: string;
	arguments: CommandArgument[];
};

export type CommandData = {
	id: CommandItem['id'];
	timestamp: number;
	type: string;
	data?: number | string;
};

export type Command = {
	item: CommandItem;
	data?: CommandData[];
};
