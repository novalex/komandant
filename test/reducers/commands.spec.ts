import { AnyAction } from 'redux';

import { CommandData, CommandItem } from '../../app/types';

import reducer, {
	initialState,
	addCommand,
	removeCommand,
	commandOutput,
	clearOutput,
} from '../../app/features/commands/slice';

const testCommand: CommandItem = {
	id: 'test-id',
	name: 'Test name',
	content: 'test content',
	arguments: [],
};

const testCommandData: CommandData = {
	id: 'test-id',
	type: 'stdout',
	timestamp: 100,
};

describe('reducer commands', () => {
	it('should handle initial state', () => {
		expect(reducer(undefined, {} as AnyAction)).toMatchSnapshot();
	});

	it('should handle unknown action type', () => {
		expect(reducer(initialState, { type: 'unknown' })).toMatchSnapshot();
	});

	it('should handle adding command', () => {
		expect(
			reducer(initialState, {
				type: addCommand,
				payload: testCommand,
			})
		).toMatchSnapshot();
	});

	it('should handle modifying existing command', () => {
		reducer(initialState, {
			type: addCommand,
			payload: testCommand,
		});

		const modifiedCommand = {
			...testCommand,
			name: 'Test name modified',
		};
		expect(
			reducer(initialState, {
				type: addCommand,
				payload: modifiedCommand,
			})
		).toMatchSnapshot();
	});

	it('should handle removing command', () => {
		const state = reducer(initialState, {
			type: addCommand,
			payload: testCommand,
		});

		expect(
			reducer(state, {
				type: removeCommand,
				payload: 'fake-test-id',
			})
		).toMatchSnapshot();

		expect(
			reducer(state, {
				type: removeCommand,
				payload: 'test-id',
			})
		).toMatchSnapshot();
	});

	it('should handle adding data', () => {
		expect(
			reducer(initialState, {
				type: commandOutput,
				payload: testCommandData,
			})
		).toMatchSnapshot();
	});

	it('should handle removing data', () => {
		const state = reducer(initialState, {
			type: commandOutput,
			payload: testCommandData,
		});

		expect(
			reducer(state, {
				type: clearOutput,
				payload: 'fake-test-id',
			})
		).toMatchSnapshot();

		expect(
			reducer(state, {
				type: clearOutput,
				payload: 'test-id',
			})
		).toMatchSnapshot();
	});
});
