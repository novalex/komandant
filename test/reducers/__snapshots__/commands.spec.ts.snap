// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer commands should handle adding command 1`] = `
Object {
  "data": Array [],
  "items": Array [
    Object {
      "arguments": Array [],
      "content": "test content",
      "id": "test-id",
      "name": "Test name",
    },
  ],
}
`;

exports[`reducer commands should handle adding data 1`] = `
Object {
  "data": Array [
    Object {
      "id": "test-id",
      "timestamp": 100,
      "type": "stdout",
    },
  ],
  "items": Array [],
}
`;

exports[`reducer commands should handle initial state 1`] = `
Object {
  "data": Array [],
  "items": Array [],
}
`;

exports[`reducer commands should handle removing command 1`] = `
Object {
  "data": Array [],
  "items": Array [
    Object {
      "arguments": Array [],
      "content": "test content",
      "id": "test-id",
      "name": "Test name",
    },
  ],
}
`;

exports[`reducer commands should handle removing command 2`] = `
Object {
  "data": Array [],
  "items": Array [],
}
`;

exports[`reducer commands should handle removing data 1`] = `
Object {
  "data": Array [
    Object {
      "id": "test-id",
      "timestamp": 100,
      "type": "stdout",
    },
  ],
  "items": Array [],
}
`;

exports[`reducer commands should handle removing data 2`] = `
Object {
  "data": Array [],
  "items": Array [],
}
`;

exports[`reducer commands should handle unknown action type 1`] = `
Object {
  "data": Array [],
  "items": Array [],
}
`;
